mozilla_version/gecko.py

```diff
@@ -528,6 +593,39 @@ class FirefoxVersion(_VersionWithEdgeCases):
         'build_number': 1,
     })
 
+    _RELEASED_EDGE_CASES_MAJOR = ({
+        'major_number': 1,
+        'minor_number': 5,
+    }, {
+        'major_number': 3,
+        'minor_number': 5,
+    }, {
+        'major_number': 3,
+        'minor_number': 6,
+    }, {
+        'major_number': 14,
+        'minor_number': 0,
+        'patch_number': 1,
+        'build_number': 1,
```

Comment:
Fun fact: product-details currently considers 14.0.1 to be both a stability version[1] and a major one[2]. This implementation will fix this and make it only a major one: 

[1] https://github.com/mozilla-releng/product-details/blob/5b34c7d73075fe31641a0c846cc191d0fa1f004a/public/1.0/firefox_history_stability_releases.json#L120
[2] https://github.com/mozilla-releng/product-details/blob/5b34c7d73075fe31641a0c846cc191d0fa1f004a/public/1.0/firefox_history_major_releases.json#L18